django/test/testcases.py

```diff
@@ -730,6 +732,23 @@ def assertWarnsMessage(self, expected_warning, expected_message, *args, **kwargs
             *args, **kwargs
         )
 
+    if not PY310:
+        @contextmanager
+        def assertNoLogs(self, logger_name, level=None):
+            """
+            Assert no logs were emitted by logger_name in wrapped command.
+
+            https://bugs.python.org/issue39385
+            """
+            if isinstance(level, int):
+                level = logging.getLevelName(level)
+            try:
+                with self.assertLogs(logger_name, level):
+                    yield
+            except AssertionError as e:
+                msg = e.args[0]
+                self.assertEqual(msg, 'no logs of level %s or higher triggered on %s' % (level, logger_name))
```

Comment:
Do we need to check this message? `AssertionError` means that no logs were logged (with at least the given level), so I would change this to: 
```suggestion
                pass
```
and add `else`:
```python
try:
    with self.assertLogs(logger_name, level) as cm:
        yield
except AssetionError:
    pass
else:
    self.fail(f'Unexpected logs found: {cm.output!r}')
```

Reply:
I think it’s good practice to verify what assertion failed. If the code in the context manager raises an `AssertionError`, I would like the original failure to be raised.

However, the previous version of the patch was very flawed. The suggested test cases showed that logs emitted on the logger would not cause the assertion to fail. I rewrote the assertion, and added `test_does_not_hide_other_failures`.